backends-common/postgres/src/main/java/org/apache/james/backends/postgres/PostgresTableManager.java

```diff
@@ -36,6 +36,36 @@
 public class PostgresTableManager implements Startable {
     public static final int INITIALIZATION_PRIORITY = 1;
     private static final Logger LOGGER = LoggerFactory.getLogger(PostgresTableManager.class);
+    private static final String UPDATE_INDEX_FUNCTION_NAME = "add_column_to_table_indexes";
+    private static final String UPDATE_INDEX_FUNCTION = "CREATE OR REPLACE FUNCTION add_column_to_table_indexes(table_name text, new_column_name text)\n" +
+                                            "RETURNS VOID AS\n" +
+                                            "$$\n" +
+                                            "DECLARE\n" +
+                                            "    index_record record;\n" +
+                                            "    index_definition text;\n" +
+                                            "    column_list_string text;\n" +
+                                            "    column_list text[];\n" +
+                                            "BEGIN\n" +
+                                            "    FOR index_record IN \n" +
+                                            "        SELECT indexname\n" +
+                                            "            FROM pg_indexes\n" +
+                                            "            WHERE tablename = table_name\n" +
+                                            "    LOOP\n" +
+                                            "        SELECT pg_get_indexdef(indexrelid)\n" +
+                                            "            INTO index_definition\n" +
+                                            "            FROM pg_index\n" +
+                                            "            WHERE indexrelid = index_record.indexname::regclass;\n" +
+                                            "        column_list_string := substring(index_definition FROM '\\((.*)\\)');\n" +
+                                            "        column_list := string_to_array(column_list_string, ', ');\n" +
+                                            "        IF NOT column_list @> ARRAY[new_column_name] then\n" +
+                                            "            index_definition := regexp_replace(index_definition, '\\)$', ', ' || new_column_name || ')');\n" +
+                                            "            EXECUTE 'DROP INDEX IF EXISTS ' || index_record.indexname;\n" +
```

Comment:
As discussed yesterday, I would prefer to control the indexes within our code at the index creation phase rather than control it at Postgres DB (without much app context): creating the indexes (at James startup) then droping them then creating new altered indexes.

Hard to unit test the PL/SQL function BTW. I would like to see at least some tests (the cases I concern about) e.g.:
- Should not DROP primary key and unique indexes
- Should not DROP correct RLS indexes and recreate them again upon every James startup:
  - When data is big in Postgres. I believe recreating the indexes has a not cheap fee.
  - Scenario: If 1 node James restarting or being spawned, that would likely affect index(ex) which disrupts query performance of other James nodes

Reply:
- Should not DROP primary key and unique indexes -----> I noticed that when Postgres create index for primary key or unique constraint, it always put 'key' at the end of index name. I can use this to ignore indexes for primary key and unique constraint.
- Should not DROP correct RLS indexes and recreate them again upon every James startup -------> In fact, recreating index only run once. Index is recreated only when it does not contain RLS column.

Reply:
> t always put 'key' at the end of index name. I can use this to ignore indexes for primary key and unique constraint.

Edge case: nothing can prevent an application index could end with the "key" postfix -> then that index would not be altered to have the `domain` column.

But I won't be more picky.